src/sparseml/pytorch/utils/exporter.py

```diff
@@ -662,6 +663,36 @@ def _save_label_to_class_mapping(
     )
 
 
+def _flatten_qparams(model: onnx.ModelProto):
+    # transforms any QuantizeLinear/DequantizeLinear that have
+    # zero_point/scale with shapes `(1,)` into shape `()`
+    graph = ONNXGraph(model)
+
+    inits_to_flatten = set()
+
+    for node in model.graph.node:
+        if node.op_type in ["QuantizeLinear", "DequantizeLinear"]:
+            # scale is required
+            scale_init = graph.get_init_by_name(node.input[1], allow_optional=False)
+            if list(scale_init.dims) == [1]:
+                inits_to_flatten.add(node.input[1])
+
+                # zero_point is optional AND shape must match
+                # scale. so if scale is (1,), then so will zero point
+                if len(node.input) == 3:
+                    inits_to_flatten.add(node.input[2])
+
+    for i, init in enumerate(model.graph.initializer):
+        if init.name not in inits_to_flatten:
+            continue
+        a = numpy_helper.to_array(init)
+        assert a.shape == (1,)
```

Comment:
why are we asserting here? if the shape were say (8,) for a channel quantized model this may crash

Reply:
the only things that get added to inits_to_flatten should have this shape (per the `list(init.dims) == [1]` above)